spec/jobs/trs_update_mentor_details_job_spec.rb

```diff
@@ -0,0 +1,115 @@
+require "rails_helper"
+
+RSpec.describe TRSUpdateMentorDetailsJob, type: :job do
+  describe "#perform" do
+    context "when the mentor has a valid trn" do
+      let(:mentor) { create(:mentor, first_name: "Joe", last_name: "Bloggs", trn: "1234567") }
+
+      before do
+        mentor
+        success_stub_request
+      end
+
+      it "updates the mentors first_name and last_name attributes" do
+        expect { described_class.perform_now(mentor) }.to change {
+          mentor.reload.first_name
+        }.from("Joe").to("Judith").and change {
+          mentor.reload.last_name
+        }.from("Bloggs").to("Chicken")
+      end
+    end
+
+    context "when the mentor has an invalid trn" do
+      let(:mentor) { create(:mentor, first_name: "Joe", last_name: "Bloggs", trn: "2222222") }
+
+      before do
+        mentor
+        failure_stub_request
```

Comment:
Could also expect Sentry to receive capture_exception to test that we are manually sending it to sentry.